From 80424e04b396c3cc7f7cabd3c7bc782139a51ac1 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Thu, 23 Jul 2015 09:21:03 +0800
Subject: [PATCH] hdmi:rk3288/rk3368: enable PHY SVSRET mode while in
 power-down mode.

While in power-down mode, the HDMI Tx PHY can be additionally set in
a lower mode of consumption by enabling the SVSRET mode, achieved by
asserting the SVSRET_MODEZ signal low.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 .../video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c b/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
index 5ca7c1a15eeb..49495bffc18e 100755
--- a/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
@@ -318,9 +318,10 @@ static const struct phy_mpll_config_tab *get_phy_mpll_tab(
 static void rockchip_hdmiv2_powerdown(struct hdmi_dev *hdmi_dev)
 {
 	hdmi_msk_reg(hdmi_dev, PHY_CONF0,
-		     m_PDDQ_SIG | m_TXPWRON_SIG | m_ENHPD_RXSENSE_SIG,
+		     m_PDDQ_SIG | m_TXPWRON_SIG |
+		     m_ENHPD_RXSENSE_SIG | m_SVSRET_SIG,
 		     v_PDDQ_SIG(1) | v_TXPWRON_SIG(0) |
-		     v_ENHPD_RXSENSE_SIG(1));
+		     v_ENHPD_RXSENSE_SIG(1)) | v_SVSRET_SIG(0);
 	hdmi_writel(hdmi_dev, MC_CLKDIS, 0x7f);
 }
 
-- 
2.35.3

